config/config.ts

```diff
@@ -0,0 +1,24 @@
+import { env, environment } from "../src/typeDefinitions/default.types";
+import {
+  RDS_BASE_API_URL,
+  RDS_BASE_STAGING_API_URL,
+  RDS_BASE_DEVELOPMENT_API_URL,
+} from "../src/constants/urls";
+
+const config = (env: env) => {
+  const environment: environment = {
+    production: {
+      RDS_BASE_API_URL: RDS_BASE_API_URL,
+    },
+    staging: {
+      RDS_BASE_API_URL: RDS_BASE_STAGING_API_URL,
+    },
+    default: {
+      RDS_BASE_API_URL: RDS_BASE_DEVELOPMENT_API_URL,
+    },
```

Comment:
can we use object destructuring to make the code more readable ?
like this
```javascript
const environment: environment = {
  production: { RDS_BASE_API_URL },
  staging: { RDS_BASE_STAGING_API_URL },
  default: { RDS_BASE_DEVELOPMENT_API_URL },
};
```

Reply:
why is this being nested inside another object? can we not have something like this: 

``` typescript
const env: environment = {
    production: RDS_BASE_API_URL, 
    staging: RDS_BASE_STAGING_API_URL,
    default: RDS_BASE_DEVELOPMENT_API_URL
}
````

Reply:
> why is this being nested inside another object? can we not have something like this:

There will be more environment keys in future therefore I have made it this way

Reply:
for that, you can have a config folder inside the application which has config for each env and then import that config here and use it something like: 

```typescript
import { prod_env } from './config/prod_env'
import { staging_env } from './config/staging_env'
import { dev_env } from './config/dev_env'

const env: environment = { 
    production: prod_env, 
    staging: staging_env, 
    default: dev_env
}
```

Reply:
That's a great idea but for a single property as of now,  I think making a separate file is redundant as of now. We can consider this when our environment object grows large enough and should have atleast 4 to 5 properties

Reply:
If so why not do it like this: 

```typescript
const env: environment = {
    production: RDS_BASE_API_URL, 
    staging: RDS_BASE_STAGING_API_URL,
    default: RDS_BASE_DEVELOPMENT_API_URL
}
```

Reply:
When calling the we will be getting the value by those keys right and we are expecting a whole config which will have keys right not only the `BASE_API_URL`

As per current scenario, we will access the API_URL like this.
```JS
config(env).RDS_BASE_API_URL // We clearly know what we are getting here
```

but as per your suggestion it will look something like this
```JS
config(env) // We know we are getting the config but not clear what exactly it is
```

Hence, code readability improves for the 1st approach

Reply:
Valid point, in that case should we not have a consistent structure across configs ? 

Example:

```typescript
const env = {
   production:{
        backendBaseUrl : <value>
   },
   staging:{
        backendBaseUrl : <value>
   }
}

// The config can then be used as 
config(env).backendBaseUrl 
```

Reply:
It consistent right -> `RDS_BASE_API_URL` and it's constant therefore following snake case (all caps)

Reply:
My bad did not see it

Reply:
Pheww !!, a lot of discussion, can you please approve it, if all looks good to you?

Reply:
Thanks for approving the PR